### PR TITLE
Add `pMapSkip`

### LINKED ISSUE #38: Allow leaving an empty value
For example:

```js
await pMap(["example.com", "notawebsite.invalid", "google.com"], async url => {
	try {
		return await got(url)
	} catch {
		return pMap.hole()
	}
})
//=> ["<example.com contents>", "<google.com contents>"]
```

### PR DESCRIPTION
**Pull request by Richienb**
Fixes #38

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -58,4 +58,35 @@ export default function pMap<Element, NewElement>(
...
+		return requestUrl;
+	} catch {
+		return skip
+	}
+};
+
+const result = await pMap(sites, mapper, {concurrency: 2});
+
+console.log(result);
+//=> ['https://sindresorhus.com/', 'https://avajs.dev/', 'https://github.com/']
```
Comment (papb) [inline index.d.ts:89]: I think it would be better if the example actually skipped one of the inputs.

**Comment by sindresorhus** [comment]
I'm not sold on calling it just `skip` when it's a named export. Either a more verbose name or put it on `pMap`. I think I would prefer the former.

**Comment by Richienb** [comment]
> I'm not sold on calling it just `skip` when it's a named export. Either a more verbose name or put it on `pMap`. I think I would prefer the former.

Maybe `pMapSkip`?

File: readme.md [hunk 2]
```diff
@@ -72,6 +72,36 @@ Default: `true`
 
 When set to `false`, instead of stopping when a promise rejects, it will wait for all the promises to settle and then reject with an [aggregated error](https://github.com/sindresorhus/aggregate-error) containing all the errors from the rejected promises.
 
+### pMap.skip
```
Comment (sindresorhus) [inline readme.md:75]: This is not correct, since it's a named export.

**Comment by sindresorhus** [comment]
> Maybe pMapSkip?

Yeah, I cannot think of anything better.

File: index.d.ts [hunk 3]
```diff
@@ -58,4 +58,36 @@ export default function pMap<Element, NewElement>(
...
 	mapper: Mapper<Element, NewElement>,
 	options?: Options
-): Promise<NewElement[]>;
+): Promise<Array<Exclude<NewElement, typeof pMapSkip>>>;
+
+/**
+Return this value from a `mapper` function to skip adding a value in the returned array.
+
+@example
+```js
```
Comment (sindresorhus) [inline index.d.ts:67]:
```suggestion
```
```

File: readme.md [hunk 4]
```diff
@@ -72,6 +72,36 @@ Default: `true`
 
 When set to `false`, instead of stopping when a promise rejects, it will wait for all the promises to settle and then reject with an [aggregated error](https://github.com/sindresorhus/aggregate-error) containing all the errors from the rejected promises.
 
+### pMapSkip
+
+Return this value from a `mapper` function to skip adding a value in the returned array.
```
Comment (sindresorhus) [inline readme.md:77]:
```suggestion
Return this value from a `mapper` function to skip including the value in the returned array.
```

**Comment by sindresorhus** [comment]
Can you fix the merge conflict?

**Comment by Richienb** [comment]
@sindresorhus Could you release this for now?

**Comment by sindresorhus** [comment]
https://github.com/sindresorhus/p-map/releases/tag/v5.1.0

**Pull request merged**